### PR TITLE
React 16 testing dependency logic update

### PR DESCRIPTION
**Pull request by aaronlademann-wf**
## Motivation
1. We modify all top-level renders to have a root of `ErrorBoundary` - but we are not forcing the addition of the `over_react` dependency.
2. We need consumers to pull in the alpha release of `web_skin_dart` for testing purposes - which means we need these alpha releases of `react` / `over_react` to be specified as `dependencies`, not `dependency_overrides` in order to automatically pull in WSD as well for the testing branch.

## Changes
1. https://github.com/Workiva/over_react_codemod/pull/60/commits/64f312541f6a0f6b9795a1377c4588719bc9f635
2. https://github.com/Workiva/over_react_codemod/pull/60/commits/34a1d45bdc6845fd06dc34efdcfa0858a6570daa

#### Release Notes
Update React 16 codemod dependency update logic

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: @greglittlefield-wf @kealjones-wk @joebingham-wk 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        - [ ] Verify that in https://github.com/Workiva/dart_storybook/pull/117, the deps updated as expected, and that the `pubspec.lock` file generated by the CI build includes `web_skin_dart 2.42.0-alpha.0`.
          (I created that PR using these changes)
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/executables/react16_dependency_override_update.dart [hunk 1]
```diff
@@ -27,14 +32,23 @@ Then, review the the changes and commit.
...
+      VersionConstraint.parse(overReactVersionRangeForTesting);
+
   final pubspecYamlQuery = FileQuery.dir(
     pathFilter: (path) => p.basename(path) == 'pubspec.yaml',
     recursive: true,
   );
 
   exitCode = runInteractiveCodemod(
     pubspecYamlQuery,
-    Ignoreable(DependencyOverrideUpdater()),
```
Comment (greglittlefield-wf) [inline lib/src/executables/react16_dependency_override_update.dart]: Is `DependencyOverrideUpdater` not used anymore? Can we remove it?

File: lib/src/executables/react16_upgrade.dart [hunk 2]
```diff
@@ -51,7 +51,7 @@ void main(List<String> args) {
     AggregateSuggestor([
       PubspecReactUpdater(reactVersionConstraint, shouldAddDependencies: false),
       PubspecOverReactUpgrader(overReactVersionConstraint,
-          shouldAddDependencies: false)
+          shouldAddDependencies: true),
```
Comment (greglittlefield-wf) [inline lib/src/executables/react16_upgrade.dart]: Why was this changed?

File: lib/src/react16_suggestors/pubspec_react_upgrader.dart [hunk 3]
```diff
@@ -50,8 +50,11 @@ class PubspecReactUpdater implements Suggestor {
         if (shouldUpdateVersionRange(
             constraint: constraint, targetConstraint: targetConstraint)) {
           // Wrap the new constraint in quotes if required.
-          var newValue =
-              generateNewVersionRange(constraint, targetConstraint).toString();
+          var newValue = targetConstraint.toString().contains('-alpha') ||
```
Comment (greglittlefield-wf) [inline lib/src/react16_suggestors/pubspec_react_upgrader.dart]: Can we add a simple test case for this new logic?

File: lib/src/executables/react16_dependency_override_update.dart (hunk 1 above)
Comment (kealjones-wk) [reply lib/src/executables/react16_dependency_override_update.dart]: same question...

Comment (aaronlademann-wf) [reply lib/src/executables/react16_dependency_override_update.dart]: We could remove it - but if we ever needed to do an update that involved dependency_overrides - we'd have to rewrite it.

File: lib/src/executables/react16_upgrade.dart (hunk 2 above)
Comment (aaronlademann-wf) [reply lib/src/executables/react16_upgrade.dart]: Item 1 in the PR description @greglittlefield-wf

Comment (greglittlefield-wf) [reply lib/src/executables/react16_upgrade.dart]: :doh: 👍

File: lib/src/react16_suggestors/pubspec_react_upgrader.dart (hunk 3 above)
Comment (aaronlademann-wf) [reply lib/src/react16_suggestors/pubspec_react_upgrader.dart]: done

**Comment by greglittlefield-wf** [comment]
+10

**Comment by aaronlademann-wf** [comment]
@Workiva/release-management-pp
